### PR TITLE
ENH: Make service more production ready

### PR DESCRIPTION
**Pull request by arkilic**
files for testing and potential offline work

### CODE REVIEW

**Comment by klauer** [comment]
Not too familiar with the API so take this with a grain of salt, but would breaking it up into `SampleReference` and `LocalSampleReference(SampleReference)` be an option?

**Comment by arkilic** [comment]
@klauer actually, I did not do any work and just pushed this bc I had the same dilemma

@dchabot that is coming

Keep the suggestions coming

File: amostra/client/commands.py [hunk 1]
```diff
@@ -24,24 +24,47 @@ def __init__(self, sample_list=[], host=conf.conn_config['host'],
...
+#                              data=domt)
+#            r.raise_for_status()
 
+    @property
+    def host(self):
+        return self.__host
+    
+    @host.setter
+    def host(self, new_host):
+        self.__host = new_host
```
Comment (klauer) [inline amostra/client/commands.py]: If there's no logic here, probably not worth having properties for this (as the getter/setter are identical to just accessing '.__host'

Also, probably not necessary to munge the name with double-underscore, single would suffice and not make it a pain to poke during debugging

File: amostra/client/commands.py [hunk 2]
```diff
@@ -337,3 +373,114 @@ def get_schema(self):
...
+        with open(self._samp_fname, 'r') as fp:
+            print(fp)
+            local_payload = ujson.load(fp)
+        print(local_payload, type(local_payload))
+        
+        
+class LocalRequestReference:
+    def __init__(self, top_dir=conf.local_conn_config['top']):
+        self.top_dir = top_dir        
+        try:
```
Comment (klauer) [inline amostra/client/commands.py]: Seems like you duplicate this json loading logic in a few places, might want to break it out into a separate function

File: amostra/client/commands.py (hunk 1 above)
Comment (arkilic) [inline amostra/client/commands.py]: Fair enough. This isa leftover from my learning attempt lol

**Comment by arkilic** [comment]
@hhslepicka take a look at d32cb13. I think this addresses the issue we talked about yesterday. It also includes the bulk insert capability (implemented on client side since it seems to be faster for <1k samples).

**Comment by arkilic** [comment]
@hhslepicka this fixes the issues with bulk insert. BTW, I think we need to add the defaulting behavior to the server side if you are building web interfaces on top of this 💭

**Comment by hhslepicka** [comment]
@arkilic I agree. It must be on the server side.

**Comment by hhslepicka** [comment]
:+1: Looks good to me!

**Pull request merged**